src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -106,6 +110,83 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 // ReserveMemoryString is amount of memory that we reserve for other processes than redpanda in the container
 const ReserveMemoryString = "1M"
 
+func (r *Cluster) validateAdminListeners() field.ErrorList {
+	var allErrs field.ErrorList
+	externalAdmin := r.AdminAPIExternal()
+	targetAdminCount := 1
+	if externalAdmin != nil {
+		targetAdminCount = 2
+	}
+	if len(r.Spec.Configuration.AdminAPI) != targetAdminCount {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec").Child("configuration").Child("adminApi"),
+				r.Spec.Configuration.AdminAPI,
+				"need exactly one internal API listener and up to one external"))
+	}
+
+	if externalAdmin != nil && externalAdmin.Port != 0 {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec").Child("configuration").Child("adminApi"),
+				r.Spec.Configuration.AdminAPI,
+				"external admin listener cannot have port specified"))
+	}
+	return allErrs
+}
+
+func (r *Cluster) validateKafkaListeners() field.ErrorList {
+	var allErrs field.ErrorList
+	if len(r.Spec.Configuration.KafkaAPI) == 0 {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec").Child("configuration").Child("kafkaApi"),
+				r.Spec.Configuration.KafkaAPI,
+				"need at least one kafka api listener"))
+	}
+
+	var external *KafkaAPIListener
+	for i, p := range r.Spec.Configuration.KafkaAPI {
+		if p.External.Enabled {
+			if external != nil {
+				allErrs = append(allErrs,
+					field.Invalid(field.NewPath("spec").Child("configuration").Child("kafkaApi"),
+						r.Spec.Configuration.KafkaAPI,
+						"only one kafka api listener can be marked as external"))
+			}
+			external = &r.Spec.Configuration.KafkaAPI[i]
+			if external.Port != 0 {
+				allErrs = append(allErrs,
+					field.Invalid(field.NewPath("spec").Child("configuration").Child("kafkaApi"),
+						r.Spec.Configuration.KafkaAPI,
+						"external kafka api listener cannot have port specified, it's autogenerated"))
+			}
+		}
+	}
+
+	// for now only one listener can have TLS to be backward compatible with v1alpha1 API
+	foundListenerWithTLS := false
+	for i, p := range r.Spec.Configuration.KafkaAPI {
```

Comment:
Could we have a TLS check for AdminAPI also?

Reply:
Yes! will add that, thanks